viewing_party/party.py

```diff
@@ -1,23 +1,166 @@
-# ------------- WAVE 1 --------------------
+from pprint import pprint
+#HELPER FUNCTION
+def _get_movies_watched_by_friends(user_data):
+    movies_watched_by_friends = []
+    
+    for friend in user_data["friends"]: #[{}]
+        for friend_movie in friend["watched"]: #{}
+            if friend_movie in movies_watched_by_friends:
+                continue
+            else:
+                movies_watched_by_friends.append(friend_movie)
+
+    return movies_watched_by_friends
+
+def _get_movies_not_watched_user(user_data):
+    movies_not_watched_user = []
+    movies_watched_by_friends = _get_movies_watched_by_friends(user_data) 
+
+    for movie in user_data["watched"]:
+        if movie not in movies_watched_by_friends:
+            movies_not_watched_user.append(movie)
+    return  movies_not_watched_user
+
 
+# ------------- WAVE 1 --------------------
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None 
+
+    return {"title":title, "genre":genre, "rating":rating}
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
 
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        watchlist_title = movie["title"]
+        if title == watchlist_title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)

```

Comment:
To avoid any potential issues modifying a list that's being iterated on, it's a good idea to `break` after making that modification.